src/stream/src/executor/hash_join.rs

```diff
@@ -829,6 +829,17 @@ impl<K: HashKey, S: StateStore, const T: JoinTypePrimitive> HashJoinExecutor<K,
 
             if let Some(rows) = &matched_rows {
                 join_matched_join_keys.observe(rows.len() as _);
+                if rows.len() > 10000 {
+                    tracing::debug!(target: "hash_join_amplification",
+                        matched_rows_len = rows.len(),
+                        update_table_id = side_update.ht.table_id(),
+                        match_table_id = side_match.ht.table_id(),
+                        join_key = ?key,
```

Comment:
We can actually get a decoded value by using `row` and `join_key_indices`.

Reply:
Cool, let me have a try later

Reply:
Follow up in future PR

Reply:
Updated https://github.com/risingwavelabs/risingwave/pull/16799/commits/977d15df8eae09f827df0619124a357314ce2940